### ISSUE TITLE
Mythos Rusted Artifact + Challenge mode?

### ISSUE THREAD

**Issue by Pricyprov** [SotMSteamMods/CauldronMods#1543]
**Describe the bug**
Rusted Artifact is indestructible and immune to damage even when a (blue eye) card is on top of the deck. I was playing on Ultimate mode when the bug occurred, and I tried to recreate the bug with standard Mythos but was unable to, so I believe it could be some unintended interaction between the challenge mode and Rusted Artifact

**What does it affect**
Card(s): Rusted Artifact
Deck(s): Mythos

**To Reproduce**
1. Ultimate Mythos
2. Have Rusted Artifact in play and a blue card on top of the villain deck
3. attempt to damage rusted artifact

**Expected behavior**
Rusted Artifact should not be immune to damage and destruction while the top card of the villain deck is blue on challenge mode.

**Screenshots**
![image](https://user-images.githubusercontent.com/75911675/145925247-abb65f27-5d68-4b59-9753-b20962b2aa34.png)

![image](https://user-images.githubusercontent.com/75911675/145925270-976dcc40-3818-4a83-bf33-53e646e67474.png)

**Comment by wrhyme29** [comment]
This is not a bug - rather Mythos was flipped on Ultimate. His advanced text on his flipped side reads "Activate all red effects" So even though the top card was blue, Rusted Artifact was still immune and indestructible.

Closing this issue, but thanks for reporting!

**Comment by Pricyprov** [comment]
I ran into the same issue again today while playing Ultimate Mythos, and it definitely occurred while on the front side of his character card, so it shouldn't be the advanced text that is causing the issue. The images I submitted with the bug report also show Mythos' character card in the target selection bar and he hadn't flipped yet then either. Maybe when Rusted Artifact is played it checks the top card and becomes indestructible if the checked card is red or green, but isn't losing that indestructibility when the card is no longer red or green?
